qa/tasks/ceph_manager.py

```diff
@@ -1360,7 +1360,9 @@ def raw_cluster_cmd_result(self, *args, **kwargs):
         kwargs['args'], kwargs['check_status'] = args, False
         return self.run_cluster_cmd(**kwargs).exitstatus
 
-    def run_ceph_w(self, watch_channel=None):
+    # XXX: setting shell to True by default is not good because killing the
+    # process ceph -w and running stdin.wait() on it leads to a crash.
+    def run_ceph_w(self, watch_channel=None, shell=False):
```

Comment:
This looks like a hack. What's the crash? Why does not using a shell help?

Reply:
Yes, probably, there could be a better fix. The crash was silent everytime, there was no entry in log. See this one for example - https://jenkins.ceph.com/view/mgr-dashboard/job/ceph-api/8313/consoleFull#-85362248744e9240e-b50a-4693-bac0-8a991bac86ac.

Reply:
The crash should happen on Ceph API tests for this PR in a couple of hours - https://github.com/ceph/ceph/pull/38471

Reply:
@batrick 

> Since it works with shell=False, I am expecting the issue must be somewhere in/around subprocess.Popen.communicate() or subprocess.Popen.poll(). Setting shell=True gives Python lesser control in some way, I guess.

This turned out to be true - https://jenkins.ceph.com/job/ceph-api/8337/consoleFull#-85362248744e9240e-b50a-4693-bac0-8a991bac86ac

Reply:
https://github.com/ceph/ceph/pull/38471#discussion_r538572712

Reply:
This `shell` argument is unused. You need to explain why.

Reply:
@batrick In my understanding, teuthology always runs every command with `shell` set to `True`. Since this problem doesn't occur with teuthology, I think looking at the ticket is self-explanatory (especially now, since I updated the ticket). Please let me know if the comment here still needs an upgrade.

Reply:
Updated the comment too.